master/include/logic/stateLogic.hpp

```diff
@@ -1,104 +1,109 @@
-
+            DigitalSender::enterOffState();
             state = AS_OFF;
             break;
 
         case AS_OFF:
+
             // If manual driving checkup fails, the car can't be in OFF state, so it goes back to MANUAL
             if (_checkupManager.manualDrivingCheckup()) {
-                //TODO: OFF -> MANUAL Operations
+                DigitalSender::enterManualState();
                 state = AS_MANUAL;
                 break;
             }
+
             if (_checkupManager.offCheckup())
                 break;
 
-            // TODO: OFF -> READY Operations
+            DigitalSender::enterReadyState();
             state = AS_READY;
             break;
 
         case AS_READY:
             if (_checkupManager.emergencyCheckup()) {
-                // Emergency, go to emergency state
-                /* TODO:
-                ASSI to blue flashing
-                EBS to Active
-                TS to OFF (open sdc)
-                Buzzer to on for 8-10s
-                */
+                performEmergencyOperations();
                 state = AS_EMERGENCY;
                 break;
             }
             if (_checkupManager.r2dCheckup()) {
                 // Not ready, do nothing
                 break;
             }
-        /* TODO:
-        Ready to drive, go to driving state
-        ASSI to yellow flashing
-        EBS to inactive
-        */
+
+            DigitalSender::enterDrivingState();
             state = AS_DRIVING;
             break;
         case AS_DRIVING:
             if (_checkupManager.emergencyCheckup()) {
-                // Emergency, go to emergency state
+                performEmergencyOperations();
                 state = AS_EMERGENCY;
                 break;
             }
-            if (_checkupManager.missionFinishedCheckup()) {
+            if (_checkupManager.drivingCheckup()) {
+                // Mission not complete, do nothing
                 break;
             }
+
+            DigitalSender::enterFinishState();
+            state = AS_FINISHED;
             break;
         case AS_FINISHED:
             if (_checkupManager.resTriggered()) {
-                // TODO: perform necessary actions to enter AS_EMERGENCY
+                performEmergencyOperations();
                 state = AS_EMERGENCY;
                 break;
             }
             if (_checkupManager.missionFinishedCheckup()) {
                 break;
             }
-        // TODO: perform necessary actions to enter AS_OFF
+
+            DigitalSender::enterOffState();
             state = AS_OFF;
             break;
         case AS_EMERGENCY:
             if (_checkupManager.emergencySequenceComplete()) {
                 break;
             }
-        // TODO: perform necessary actions to enter AS_OFF
+
+            DigitalSender::enterOffState();
             state = AS_OFF;
             break;
         default:
             break;
     }
 }
+
+inline void ASState::performEmergencyOperations() {
```

Comment:
What is this function for?

Reply:
avoiding repeating the emergency operations 3 times, just call the function

Reply:
but there is only one statement inside of it